src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/OsxIPv4GlobalStatistics.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.Net.NetworkInformation
+{
+    internal class OsxIPv4GlobalStatistics : IPGlobalStatistics
+    {
+        private readonly long _outboundPackets;
+        private readonly long _outputPacketsNoRoute;
+        private readonly long _cantFrags;
+        private readonly long _datagramsFragmented;
+        private readonly long _packetsReassembled;
+        private readonly long _totalPacketsReceived;
+        private readonly long _packetsDelivered;
+        private readonly long _packetsDiscarded;
+        private readonly long _packetsForwarded;
+        private readonly long _badAddress;
+        private readonly long _badHeader;
+        private readonly long _unknownProtos;
+        private readonly int _defaultTtl;
+        private readonly bool _forwarding;
+        private readonly int _numInterfaces;
+        private readonly int _numIPAddresses;
+        private readonly int _numRoutes;
+
+        public OsxIPv4GlobalStatistics()
+        {
+            Interop.Sys.IPv4GlobalStatistics statistics;
+            if (Interop.Sys.GetIPv4GlobalStatistics(out statistics) == -1)
+            {
+                throw new NetworkInformationException((int)Interop.Sys.GetLastError());
+            }
+
+            _outboundPackets = (long)statistics.OutboundPackets;
+            _outputPacketsNoRoute = (long)statistics.OutputPacketsNoRoute;
+            _cantFrags = (long)statistics.CantFrags;
+            _datagramsFragmented = (long)statistics.DatagramsFragmented;
+            _packetsReassembled = (long)statistics.PacketsReassembled;
+            _totalPacketsReceived = (long)statistics.TotalPacketsReceived;
+            _packetsDelivered = (long)statistics.PacketsDelivered;
+            _packetsDiscarded = (long)statistics.PacketsDiscarded;
+            _packetsForwarded = (long)statistics.PacketsForwarded;
+            _badAddress = (long)statistics.BadAddress;
+            _badHeader = (long)statistics.BadHeader;
+            _unknownProtos = (long)statistics.UnknownProtos;
+            _defaultTtl = statistics.DefaultTtl;
+            _forwarding = statistics.Forwarding == 1;
+
+            var interfaces = (UnixNetworkInterface[])NetworkInterface.GetAllNetworkInterfaces();
+
+            // PERF: In native shim, use sysctl: net.link.generic.system.ifcount = number of network interfaces
```

Comment:
I think we should always use TODO for things that are not already done. 
Reading this I'm not sure if you're calling out that something has already been done for perf reasons or if there is a potential perf improvement.

I'd do something like `//TODO: #1234 PERF - ...`